app/controllers/users_controller.rb

```diff
@@ -344,6 +345,46 @@ def logon_after_password_reset
     @success = I18n.t(message)
   end
 
+  def admin_login
+
+    unless SiteSetting.enable_sso && !current_user
+      return redirect_to path("/")
+    end
+
+    if request.put?
+      RateLimiter.new(nil, "admin-login-hr-#{request.remote_ip}", 6, 1.hour).performed!
+      RateLimiter.new(nil, "admin-login-min-#{request.remote_ip}", 3, 1.minute).performed!
+
+      user = User.find_by(email: params[:email])
```

Comment:
It might be better to filter the admin/system out via your your query instead of app code:

`user = User.where(email: params[:email], admin: true).where.not(id: Discourse::SYSTEM_USER_ID).first`